### DISCUSSION TITLE
Split/join list-like construct

### DISCUSSION THREAD

**Discussion by musjj** [helix-editor/helix#8507] [General]
This will allow you to split constructs like:

```
func_call(arg1, arg2, arg3)

let a = [ "foo", "bar", "baz" ]
```

Into:

```
func_call(
  arg1,
  arg2,
  arg3
)

let a = [
  "foo",
  "bar",
  "baz"
]
```

And also the other way around.

Treesitter would be a perfect fit for this.

Prior art: 
- https://github.com/Wansmer/treesj
- https://github.com/bennypowers/splitjoin.nvim

**Comment by pascalkuthe** [comment]
you can already do that with multipcursor easily. `xs<space><ret>c<ret><esc>` to slip and there is already a joint command.

The only thing to improve here would be to allow better selection with TS. That would likely be something like "split at TS siblings" .

**Reply by musjj** [reply]
Yes, a treesitter solution would be superior and more generally applicable. For example, the split solution doesn't work for functions because you have to perform an extra operation to add newlines next to the brackets.
